syft/check.py

```diff
@@ -0,0 +1,70 @@
+from dpcontracts import require, ensure
+
+
+def type_hints(decorated):
+    # """If __debug__ is set to true, enforce python argument type hints
+    #
+    #   As a convention across PySyft,
+    #
+    #   Arg:
+    #       decorated (func): the function on which we want to check argument types
+    #
+    #   Returns:
+    #       decorator (func): the decorated function which checks argument types
+    #   """
+
+    def check_argument_types(args):
```

Comment:
Why not use http://mypy-lang.org/?

I'm not familiar with it but it seems to do the job.